components/places/src/places.udl

```diff
@@ -145,63 +171,48 @@ message BookmarkNode {
      * - Not provided if `child_nodes` is provided, to avoid sending more data
      *   over the FFI than necessary.
      */
-    repeated string child_guids = 9;
+    sequence<string> child_guids;
 
     /**
      * Data about folder children, in order. Only present for type =
      * `BookmarkType::Folder`.
      *
      * For performance reasons, this only is provided if it's requested.
      */
-    repeated BookmarkNode child_nodes = 10;
-
-    /**
-     * Returned by reads, and used to distinguish between the cases of
-     * "empty child_nodes because the API doesn't return children" and
-     * "empty child_nodes because this folder has no children (but
-     * we'd populate them if it had them)".
-     *
-     * Only required because you can't have `optional repeated`.
-     *
-     * Leaving this out is equivalent to false.
-     */
-    optional bool have_child_nodes = 11;
```

Comment:
UniFFI can happily represent an `Option<Vec<T>>` without ambiguity, so we don't need this silliness that is only here to work around that fact that protobuf can't. (No slight against protobuf, we're using it in a way that's not exactly aligned with its core value prop).